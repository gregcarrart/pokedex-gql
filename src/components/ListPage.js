import React from 'react';
import { Query } from 'react-apollo';
import styled from 'styled-components';
import gql from 'graphql-tag';
import ListItem from './ListItem';

export const GET_POKEMON = gql`
  {
    pokemons(first: 10) {
      id
      number
      name
      maxCP
      maxHP
      image
      types
    }
  }
`;

const HeaderTitle = styled.h1`
  text-align: center;
`;

const GridContainer = styled.div`
  display: flex;
  width: 100%;
  max-width: 1200px;
  margin: 25px auto;
  flex-flow: row wrap;
`;

const ListPage = () => (
  <Query query={GET_POKEMON}>
    {({ loading, error, data }) => {
      if (loading) return 'Loading...';
      if (error) return `Error! ${error.message}`;

      return (
        <div>
          <HeaderTitle>Pokedex</HeaderTitle>
          <GridContainer>
            { 
              data.pokemons.map(pokemon => (
                <ListItem key={pokemon.id} pokemon={pokemon} /> 
              ))
            }
          </GridContainer>
        </div>
      );
    }}
  </Query>
);
  
/**
 * @component
 */
export default ListPage;