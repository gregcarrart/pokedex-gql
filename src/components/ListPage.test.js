import React from 'react';
import renderer from 'react-test-renderer';
import { MockedProvider } from 'react-apollo/test-utils';
import { GET_POKEMON } from './ListPage';
import { BrowserRouter as Router } from 'react-router-dom';
import ListPage from './ListPage';

const mocks = [
  {
    request: {
      query: GET_POKEMON,
    },
    result: {
      data: {
        pokemon: { id: '1', name: 'Bulbasaur' },
      },
    },
  
  },
];
describe('ListPage component tests', () => {
  test('should match snapshot', () => {
    const component = renderer.create(
      <Router>
        <MockedProvider mocks={mocks} addTypename={false}>
          <ListPage />
        </MockedProvider>
      </Router>,
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
