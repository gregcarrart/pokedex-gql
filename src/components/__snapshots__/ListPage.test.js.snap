// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListPage component tests should match snapshot 1`] = `"Loading..."`;
